### PR TITLE
Code generation integration tests

### PR DESCRIPTION
**Pull request by Oleg-Melnik**
This PR includes the following things done:

1. Add new module `integration-tests`.
2. Implement test that checks various use-cases of using the `size` option.

### CODE REVIEW

File: integration-tests/src/test/kotlin/spine/protodata/hello/SizeOptionTest.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import io.spine.protodata.hello.validatePhoneCount
+import org.junit.jupiter.api.Test
+
+/**
+ * Checks various use-cases on `size` option validation.
+ */
+class `Size option test should` {
+
+    @Test
+    fun `check several options within the same message type and sub-type`() {
```
Comment (armiol) [inline integration-tests/src/test/kotlin/spine/protodata/hello/SizeOptionTest.kt:17]: Could you add some negative tests here, in the same manner as this one?

I assume we should be expecting some (particular) `RuntimeException` in such cases.

File: integration-tests/src/test/proto/spine/protodata/hello/contact.proto [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+option (type_url_prefix) = "type.protodata.hello";
+option java_multiple_files = true;
+option java_outer_classname = "ContactProto";
+option java_package = "io.spine.protodata.hello";
+
+import "google/protobuf/descriptor.proto";
+import "spine/protodata/hello/options.proto";
+
+// Test message that contains several fields with `size` option applied.
```
Comment (armiol) [inline integration-tests/src/test/proto/spine/protodata/hello/contact.proto:41]: I would remove this comment, as it still is only present for one message from several. And from being in `integration-tests/src/test/`, it is clear enough it serves some testing purpose.

File: integration-tests/src/test/proto/spine/protodata/hello/contact.proto [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  repeated Address address = 3 [(required) = true, (size).value = "element_count"];
+
+  repeated string email = 4 [(required) = true, (size).value = "element_count"];
+}
+
+message Address {
+  int32 number_of_lines = 1 [(required) = true, (min).value = "2"];
+
+  repeated string address_line = 2 [(required) = true, (size).value = "number_of_lines"];
```
Comment (armiol) [inline integration-tests/src/test/proto/spine/protodata/hello/contact.proto:55]: Let's also test positive scenarios with expressions determining `(size).value`.

File: integration-tests/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import io.spine.internal.dependency.JUnit
+import io.spine.internal.dependency.Validation
+
+dependencies {
+    // Enable field options extension
```
Comment (armiol) [inline integration-tests/build.gradle.kts:30]: Please add a period.

**Review by armiol — CHANGES_REQUESTED** [review]
@Oleg-Melnik please see my comments.

File: integration-tests/src/test/kotlin/spine/protodata/hello/SizeOptionTest.kt (hunk 1 above)
Comment (Oleg-Melnik) [reply integration-tests/src/test/kotlin/spine/protodata/hello/SizeOptionTest.kt:18]: Negative cases added, I just assumed that we have negative cases in the model tests.

**Comment by Oleg-Melnik** [comment]
@armiol PR is ready for re-review.

**Review by armiol — APPROVED** [review]
@Oleg-Melnik LGTM.

**Pull request merged**